### ISSUE TITLE
Fine-tuning de Tacotron 2 para Síntesis de Voz en Mapudungun con Variantes Regionales

### ISSUE THREAD

**Issue by SmnZin** [PIA-2024-II/Modulo-TTS#4]
Realizar una prueba inicial de fine-tuning de Tacotron 2 utilizando el dataset de una región del dataset para evaluar la viabilidad y desafíos en la generación de voz sintética en Mapudungun.

## Tareas

1) Preparación del Ambiente
- [ ] Clonar repositorio de Tacotron 2
- [ ] Configurar ambiente de desarrollo
- [ ] Verificar compatibilidad de CUDA y PyTorch
- [ ] Probar el modelo base con ejemplos incluidos en el repo

2) Preparación del Dataset

- [ ] Desarrollar script para procesar transcripciones:
      - Convertir formato original a formato Tacotron 2
      - Validar caracteres especiales del Mapudungun
      - Generar metadata.csv (si es requerido)
- [ ] Separar dataset:
      - 80% entrenamiento
      - 20% validación

3) Adaptación Inicial del Modelo

```
 # Ejemplos de parámetros a ajustar
batch_size = 8  # Reducido por tamaño de dataset
learning_rate = 1e-4 
```

- [ ] Modificar configuración para dataset pequeño:

- [ ] Adaptar preprocessador de texto para Mapudungun
- [ ] Configurar parámetros de audio según grabaciones
- [ ] Implementar checkpoints frecuentes

4) Entrenamiento Piloto

- [ ] Entrenar modelo con dataset de región escogida
- [ ] Monitorear:
       - Loss de entrenamiento
       - Loss de validación
       - Calidad de audio generado
- [ ] Guardar checkpoints cada 1000 pasos
- [ ] Documentar comportamiento del modelo

5) Evaluación modelo 

- [ ] Probar generación con:
      -5 palabras del conjunto de entrenamiento
     - 5 palabras del conjunto de validación
     - 3 palabras nuevas en Mapudungun

- [ ] Documentar resultados:
   - Calidad del audio
  - Precisión fonética
  - Problemas identificados

Material de Apoyo:
https://github.com/NVIDIA/tacotron2
https://pytorch.org/hub/nvidia_deeplearningexamples_tacotron2/
Recomiendo leer todo el hilo de Koyela Chakrabarti en medium sobre TTS
https://medium.com/@koyelac/text-to-speech-with-tacotron2-part-5-implementation-b9811154387d
